flink-runtime/src/main/java/org/apache/flink/runtime/rest/handler/job/JobExceptionsHandler.java

```diff
@@ -87,18 +87,23 @@ protected JobExceptionsInfo handleRequest(
     }
 
     @Override
-    public Collection<ArchivedJson> archiveJsonWithPath(AccessExecutionGraph graph)
+    public Collection<ArchivedJson> archiveJsonWithPath(ExecutionGraphInfo executionGraphInfo)
             throws IOException {
-        ResponseBody json = createJobExceptionsInfo(graph, MAX_NUMBER_EXCEPTION_TO_REPORT);
+        ResponseBody json =
+                createJobExceptionsInfo(executionGraphInfo, MAX_NUMBER_EXCEPTION_TO_REPORT);
         String path =
                 getMessageHeaders()
                         .getTargetRestEndpointURL()
-                        .replace(':' + JobIDPathParameter.KEY, graph.getJobID().toString());
+                        .replace(
+                                ':' + JobIDPathParameter.KEY,
+                                executionGraphInfo.getJobId().toString());
         return Collections.singletonList(new ArchivedJson(path, json));
     }
 
     private static JobExceptionsInfo createJobExceptionsInfo(
-            AccessExecutionGraph executionGraph, int exceptionToReportMaxSize) {
+            ExecutionGraphInfo executionGraphInfo, int exceptionToReportMaxSize) {
+        final ArchivedExecutionGraph executionGraph =
+                executionGraphInfo.getArchivedExecutionGraph();
```

Comment:
Why is the `ExecutionGraphInfo` storing/returning an `ArchivedExecutionGraph` instead of an `AccessExecutionGraph`?

Reply:
Fair point. I missed that when adding the `ExecutionGraphInfo` because it was replacing the `ArchivedExecutionGraph`. I changed it.